radixdlt/src/main/java/com/radixdlt/consensus/VertexStore.java

```diff
@@ -59,7 +59,7 @@ public VertexStore(
 	) {
 		this.engine = Objects.requireNonNull(engine);
 		this.counters = Objects.requireNonNull(counters);
-		this.highestQC = Objects.requireNonNull(rootQC);
+		this.highestQC.set(Objects.requireNonNull(rootQC));
```

Comment:
Would be better IMO to assign highestQC with a newly constructed AtomicReference with the reference already in it, eg `this.highestQC = new AtomicReference<>(Objects.requireNonNull(rootQC))`.

That way, if someone removes or otherwise modifies this line, the compilers `final` field initialisation will let them know that there is an issue, also a general dislike of two-phase initialisation when it can be avoided easily.